micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/UrlUtils.java

```diff
@@ -0,0 +1,41 @@
+package com.ofg.infrastructure.web.resttemplate.fluent;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+public class UrlUtils {
+
+    private static final String FIRST_PARAMETER_CONCATENATOR = "?";
+    private static final String PARAMETERS_ASSIGNMENT = "=";
+    private static final String SECOND_AND_LATER_CONCATENATOR = "&";
+
+    public static URI addQueryParametersToUri(URI uri, Map<String, Object> params) throws URISyntaxException {
+        StringBuilder urlParametersBuilder = new StringBuilder();
+        if (uri == null) {
+            throw new IllegalArgumentException("Define URL before URL parameters");
+        }
+        urlParametersBuilder.append(uri.toString());
+        String concatenator = FIRST_PARAMETER_CONCATENATOR;
+        for (String paramName : params.keySet()) {
```

Comment:
That's not too Groovy ;) Don't use `;` ;)

Also 

`if (!(parameterValue == null) && !"".equals(parameterValue.toString()))`

can be rewritten as 

`if(parameterValue)` - check Groovy Truth.

If you don't want to use Groovy Truth you can use `StringUtils.isNotBlank()`.

That is more Groovy:

```groovy
params.keySet().each {
    ....
}
```

or even
(check if the args shouldn't be the other way round)

```groovy
params.keySet().inject('') { String element, String acc ->
    validateParameters(paramName)
    String params = "$concatenator$paramName$PARAMETERS_ASSIGNMENT"
    if(parameterValue) {
       params = "$params${parameterValue.toString()}"
    }
    return params
}
```